tests_e2e/test_e2e.py

```diff
@@ -0,0 +1,64 @@
+import os
+from threading import Thread
+import pytest
+from selenium import webdriver
+import requests
+from todo_app.app import create_app
+
+def requestJson (method, url, args):
+    r = requests.request(method, url, params=args) 
+    r.raise_for_status()
+    print(r.raise_for_status())
+    return r.json()
+    
+    
+#Function to create trello board
+def createTrelloBoard(boardName):
+    url='https://api.trello.com/1/boards/'
+    params={
+            'key': os.getenv('TRELLO_KEY'),
+            'token': os.getenv('TRELLO_TOKEN'),
+            'name': boardName
+        }
+    
+    res=requestJson("POST",url, params)
+    return res
+
+
+#Function to remove trello board
+def deleteTrelloBoard(boardId):
+    url='https://api.trello.com/1/boards/'+boardId
+    res=requestJson("DELETE",url, {})
+    return res
+
+@pytest.fixture(scope='module')
+def app_with_temp_board():
+    # Create the new board & update the board id environment variable
+    board_id = (createTrelloBoard('Selenium_Test_Board'))["id"]
+    os.environ['trelloBoardID'] = board_id
+    
+    # construct the new application
+    app = create_app()
+    
+    # start the app in its own thread.
+    thread = Thread(target=lambda:app.run(use_reloader=False))
+    thread.daemon = True
+    thread.start()
+    
+    yield app
+    
+    # Tear Down
+    thread.join(1)
+    #deleteTrelloBoard(board_id)
+
+
+@pytest.fixture(scope='module')
+def driver():
+    with webdriver.Firefox() as driver:
+        yield driver
+
+
+def test_mainPage(driver):
```

Comment:
I think you're currently running this test against a locally hosted version of the app, rather than having the test spin up the app for you - does that sound right?
The reason I think that is you're not currently using the app_with_temp_board function that you defined above and that I can see is working - you need to specify it as a fixture you're using by mentioning it in the test's arguments:
```python
def test_mainPage(driver, app_with_temp_board):
    driver.get('http://localhost:5000/')
    assert driver.title == 'To-Do App'        
```

Without that your test fails for me, with it it passes (with the other twist above)